src/main/java/com/puppycrawl/tools/checkstyle/api/InitialModule.java

```diff
@@ -0,0 +1,59 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.api;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * The first module that is run as part of Checkstyle and controls its entire
+ * behavior and children.
+ * @author Richard Veach
+ */
+public interface InitialModule extends Configurable {
```

Comment:
please rename it to RootModule (no "initial" should be left in patch), in doc it is already named like this http://checkstyle.sourceforge.net/config.html#Checker and it more represent what it is.